sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLViewSuite.scala

```diff
@@ -448,19 +476,105 @@ class SQLViewSuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
     }
   }
 
+  test("Using view after change the origin view") {
+    withView("v1", "v2") {
+      sql("CREATE VIEW v1 AS SELECT id FROM jt")
+      sql("CREATE VIEW v2 AS SELECT * FROM v1")
+      withTable("jt2", "jt3") {
+        // Don't change the view schema
+        val df2 = (1 until 10).map(i => i + i).toDF("id")
```

Comment:
I suggest to twist this test case a bit.
The first part of the test case has the table jt defined as columns ("j", "i") and then the second part jt2 defined as, what you currently have, columns("i", "j") to test that the code checks the order of the columns correctly.